### PR TITLE
Preventing Clogging of Symbol Table

### PR DESCRIPTION
**Pull request by TanzHasan**
Currently we pull in stuff from Asm-Generics such as __u16 and also use the insides of functions. Additionally for headers only the function name matters so we don't need to traverse the whole function.

### CODE REVIEW

File: lib/graph_utils.py [hunk 1]
```diff
@@ -12,17 +13,29 @@ def visit_node(node, depth = 0):
...
+
+            if node.kind in {
+                CursorKind.STRUCT_DECL, CursorKind.ENUM_DECL, CursorKind.TYPEDEF_DECL,
+                CursorKind.CLASS_DECL, CursorKind.FUNCTION_DECL,
+                CursorKind.MACRO_DEFINITION, CursorKind.FIELD_DECL,
+            }:
+                if filename.name in str(node.location.file):
+                    symbols.add(name)
+
+            if node.kind in { CursorKind.FUNCTION_DECL }:
```
Comment (bwendling) [inline lib/graph_utils.py:33]: This seems strange. Why not just:

```
if node.kind == CursorKind.FUNCTION_DECL:
```

File: lib/graph_utils.py [hunk 2]
```diff
@@ -12,17 +13,29 @@ def visit_node(node, depth = 0):
...
 
         else:
             name = node.spelling
-            symbols.add(name)
+
+            if node.kind in {
+                CursorKind.STRUCT_DECL, CursorKind.ENUM_DECL, CursorKind.TYPEDEF_DECL,
+                CursorKind.CLASS_DECL, CursorKind.FUNCTION_DECL,
+                CursorKind.MACRO_DEFINITION, CursorKind.FIELD_DECL,
+            }:
```
Comment (bwendling) [inline lib/graph_utils.py:29]: It might be good to create this set as a global frozenset. The name would be a bit more descriptive.

Comment (nickdesaulniers) [reply lib/graph_utils.py:29]: also, this hunk of the commit seems based on #2

File: lib/graph_utils.py [hunk 3]
```diff
@@ -1,7 +1,14 @@
+'''functions used in performing hierarchical agglomeration'''
 from pathlib import Path
 from typing import List, Set
 from clang.cindex import Index, CursorKind
 
+usable_types = frozenset({
+    CursorKind.STRUCT_DECL, CursorKind.ENUM_DECL, CursorKind.TYPEDEF_DECL,
+    CursorKind.CLASS_DECL, CursorKind.FUNCTION_DECL,
```
Comment (nickdesaulniers) [inline lib/graph_utils.py]: https://github.com/ClangBuiltLinux/HeaderGraphPartioning/pull/2#discussion_r1420909542

**Review by nickdesaulniers — APPROVED** [review]

**Pull request merged**